app/services/participants/change_schedule/npq.rb

```diff
@@ -42,7 +42,17 @@ def call
       attr_reader :schedule_identifier, :cohort_year
 
       def user_profile
-        user&.npq_profiles&.active_record&.includes({ npq_application: [:npq_course] })&.where('npq_courses.identifier': course_identifier)&.first
+        user
+          &.npq_profiles
+          &.active_record
+          &.includes({ npq_application: [:npq_course] })
+          &.where('npq_courses.identifier': course_identifier)
+          &.where({ npq_application: { npq_lead_provider: } })
```

Comment:
None of those `&.` is necessary. this will always return and `ActiveRecord::Relationships` which will never be nil

Reply:
good shout, I've removed all these now